pagination/tests/test_querystrings.py

```diff
@@ -0,0 +1,38 @@
+from django.http import QueryDict
+
+from .utils import RequestTestCase
+from .. import querystrings
+
+
+class TestQuerystringUpdate(RequestTestCase):
+    def setUp(self):
+        """Create a request and put some GET args in it."""
+        self.request = self.create_request()
+        self.request.GET = QueryDict('page=3&q=search_term')
+
+    def test_unchanged_querystring(self):
+        """Test that the method returns an unchanged request.GET if kwargs is empty."""
+        querystring = querystrings.querystring_update(self.request)
+
+        # Re-parse the result into a dict because the query arguments don't appear in
+        # a predictable order.
+        new_querydict = QueryDict(querystring).dict()
+        expected = {
+            'page': '3',
+            'q': 'search_term',
+        }
+        self.assertDictEqual(new_querydict, expected)
+
+    def test_querystring_update(self):
+        """Test that individual GET arguments are updated or preserved correctly."""
+        search_term = 'a_different_search_term'
+        querystring = querystrings.querystring_update(self.request, q=search_term)
+
+        # Re-parse the result into a dict because the query arguments don't appear in
+        # a predictable order.
+        new_querydict = QueryDict(querystring).dict()
+        expected = {
+            'page': '3',
+            'q': search_term,
+        }
+        self.assertDictEqual(new_querydict, expected)
```

Comment:
`assertDictEqual` doesn't do anything useful for those:

```python
from collections import OrderedDict
from unittest import TestCase


class TestEqual(TestCase):
    def setUp(self):
        self.d = {'a': 'b', 'c': 'd', 'e': 'f'}
        self.d2 = {'a': 'b', 'c': 'd'}
        self.o = OrderedDict([('a', 'b'), ('c', 'd')])

    def test_equal_d(self):
        self.assertEqual(self.d, self.d2)

    def test_dict_equal_d(self):
        self.assertDictEqual(self.d, self.d2)

    def test_equal_o(self):
        self.assertEqual(self.d, self.o)

    def test_dict_equal_o(self):
        self.assertDictEqual(self.d, self.o)
```

```python-traceback
FAIL: test_dict_equal_d (tests.test_equal.TestEqual)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/test_equal.py", line 15, in test_dict_equal_d
    self.assertDictEqual(self.d, self.d2)
AssertionError: {'c': 'd', 'e': 'f', 'a': 'b'} != {'c': 'd', 'a': 'b'}
- {'a': 'b', 'c': 'd', 'e': 'f'}
?                    ----------

+ {'a': 'b', 'c': 'd'}

======================================================================
FAIL: test_dict_equal_o (tests.test_equal.TestEqual)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/test_equal.py", line 21, in test_dict_equal_o
    self.assertDictEqual(self.d, self.o)
AssertionError: {'c': 'd', 'e': 'f', 'a': 'b'} != OrderedDict([('a', 'b'), ('c', 'd')])
- {'a': 'b', 'c': 'd', 'e': 'f'}
+ OrderedDict([('a', 'b'), ('c', 'd')])

======================================================================
FAIL: test_equal_d (tests.test_equal.TestEqual)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/test_equal.py", line 12, in test_equal_d
    self.assertEqual(self.d, self.d2)
AssertionError: {'c': 'd', 'e': 'f', 'a': 'b'} != {'c': 'd', 'a': 'b'}
- {'a': 'b', 'c': 'd', 'e': 'f'}
?                    ----------

+ {'a': 'b', 'c': 'd'}

======================================================================
FAIL: test_equal_o (tests.test_equal.TestEqual)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/test_equal.py", line 18, in test_equal_o
    self.assertEqual(self.d, self.o)
AssertionError: {'c': 'd', 'e': 'f', 'a': 'b'} != OrderedDict([('a', 'b'), ('c', 'd')])
```

As you can see, when comparing two dictionaries, the output is equivalent. When comparing a dictionary to an ordered dict, `assertDictEqual` has superfluous output with no useful extra information.